main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -265,6 +267,10 @@ object TypeReduction {
       case (Type.Cst(TypeConstructor.Native(clazz), _), Type.Cst(TypeConstructor.Str, _)) => clazz.isAssignableFrom(classOf[String])
       case (Type.Cst(TypeConstructor.Native(clazz), _), Type.Cst(TypeConstructor.BigInt, _)) => clazz.isAssignableFrom(classOf[BigInteger])
       case (Type.Cst(TypeConstructor.Native(clazz), _), Type.Cst(TypeConstructor.BigDecimal, _)) => clazz.isAssignableFrom(classOf[java.math.BigDecimal])
+      case (Type.Cst(TypeConstructor.Array, _), Type.Cst(TypeConstructor.Array, _)) =>
+        val (elmType2, elmType1) = (tpe2.typeVars.head, tpe1.typeVars.head)
```

Comment:
Why do you swap 2 and 1 here?

We don't want `typeVars` here, but rather `typeArguments`.

Can we just unpack the type arguments instead? Like `List(elmType1, rc1) = tpe1.typeArgs` That way we avoid the `.head.tail` stuff

Reply:
I am checking if the type of array1 (tpe1) is a subtype of array2 (tpe2), I just did the tuple like this since this is what was used in the match case. I initially had in mind something like: is tpe2 a supertype of tpe1 that is why, later refactoring can be envisioned.

Ah I see, thank you I will change that.

Thank you for the tip, I am not so used with Scala indeed this seems better.